controllers/awscluster_controller.go

```diff
@@ -191,21 +195,34 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	// Handle deleted clusters
 	if !awsCluster.DeletionTimestamp.IsZero() {
-		return ctrl.Result{}, r.reconcileDelete(ctx, clusterScope)
+		return r.reconcileDelete(ctx, clusterScope)
 	}
 
 	// Handle non-deleted clusters
 	return r.reconcileNormal(clusterScope)
 }
 
-func (r *AWSClusterReconciler) reconcileDelete(ctx context.Context, clusterScope *scope.ClusterScope) error {
+func (r *AWSClusterReconciler) reconcileDelete(ctx context.Context, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
 	if !controllerutil.ContainsFinalizer(clusterScope.AWSCluster, infrav1.ClusterFinalizer) {
 		clusterScope.Info("No finalizer on AWSCluster, skipping deletion reconciliation")
-		return nil
+		return reconcile.Result{}, nil
 	}
 
 	clusterScope.Info("Reconciling AWSCluster delete")
 
+	numDependencies, err := r.dependencyCount(ctx, clusterScope)
+	if err != nil {
+		clusterScope.Error(err, "error getting AWSCluster dependencies")
+		return reconcile.Result{}, err
+	}
+
+	if numDependencies > 0 {
+		clusterScope.Info("AWSCluster cluster still has dependencies - requeue needed", "dependencyCount", numDependencies)
```

Comment:
To provide a bit more context we can say what sort of dependencies we are talking about here, as they could be confused with cloud resources.
```suggestion
		clusterScope.Info("AWSCluster still has dependent CAPI resources - requeuing", "dependencyCount", numDependencies)
```

Reply:
Could we maybe list in the k/v of this log like, what CAPI resources are still depending on the AWSCluster, e.g. instead of the count report `nameSpace/name` of the resources, or would that be too much/too verbose? What do you think Alex?

Reply:
Having the list would be helpful but as it stands its consistent with the same check that is done on the EKS side. I'd say lets follow up on logging the list of resources.

Reply:
Yeah happy to keep it with just the count in this one 👍